go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+	err = clusterInstance.VtctlProcess.AddCellInfo("test_ny")
+
+	assert.Nil(t, err, "error should be Nil")
+
+	// source keyspace, with 4 tables
+	sourceShard := clusterInstance.Keyspaces[0].Shards[0]
+	sourceMasterTablet := sourceShard.Vttablets[0]
+	sourceReplicaTablet := sourceShard.Vttablets[1]
+	sourceRdOnlyTablet1 := sourceShard.Vttablets[2]
+	sourceRdOnlyTablet2 := sourceShard.Vttablets[3]
+	sourceKs := fmt.Sprintf("%s/%s", sourceKeyspace, shardName)
+
+	// master tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&sourceMasterTablet, cellj, sourceKeyspace, hostname, sourceShard.Name)
+	_ = sourceMasterTablet.VttabletProcess.CreateDB(sourceKeyspace)
+
+	// replica tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&sourceReplicaTablet, cellj, sourceKeyspace, hostname, sourceShard.Name)
+	_ = sourceReplicaTablet.VttabletProcess.CreateDB(sourceKeyspace)
+
+	// rdOnly_1 tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&sourceRdOnlyTablet1, cellj, sourceKeyspace, hostname, sourceShard.Name)
+	_ = sourceRdOnlyTablet1.VttabletProcess.CreateDB(sourceKeyspace)
+
+	// rdOnly_2 tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&sourceRdOnlyTablet2, cellj, sourceKeyspace, hostname, sourceShard.Name)
+	_ = sourceRdOnlyTablet2.VttabletProcess.CreateDB(sourceKeyspace)
+
+	// destination keyspace, with just two tables
+	destinationShard := clusterInstance.Keyspaces[1].Shards[0]
+	destinationMasterTablet := destinationShard.Vttablets[0]
+	destinationReplicaTablet := destinationShard.Vttablets[1]
+	destinationRdOnlyTablet1 := destinationShard.Vttablets[2]
+	destinationRdOnlyTablet2 := destinationShard.Vttablets[3]
+
+	// master tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&destinationMasterTablet, cellj, destinationKeyspace, hostname, destinationShard.Name)
+	_ = destinationMasterTablet.VttabletProcess.CreateDB(destinationKeyspace)
+
+	// replica tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&destinationReplicaTablet, cellj, destinationKeyspace, hostname, destinationShard.Name)
+	_ = destinationReplicaTablet.VttabletProcess.CreateDB(destinationKeyspace)
+
+	// rdOnly_1 tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&destinationRdOnlyTablet1, cellj, destinationKeyspace, hostname, destinationShard.Name)
+	_ = destinationRdOnlyTablet1.VttabletProcess.CreateDB(destinationKeyspace)
+
+	// rdOnly_2 tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&destinationRdOnlyTablet2, cellj, destinationKeyspace, hostname, destinationShard.Name)
+	_ = destinationRdOnlyTablet2.VttabletProcess.CreateDB(destinationKeyspace)
+
+	// RebuildKeyspaceGraph source keyspace
+	_ = clusterInstance.VtctlclientProcess.ExecuteCommand("RebuildKeyspaceGraph", sourceKeyspace)
+
+	// RebuildKeyspaceGraph destination keyspace
+	_ = clusterInstance.VtctlclientProcess.ExecuteCommand("RebuildKeyspaceGraph", destinationKeyspace)
```

Comment:
catch and assert.Nil for error

Reply:
Done. Refer [PR46](https://github.com/planetscale/vitess/pull/46)